vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -902,7 +902,12 @@ def submit_property_metrics(
             )
             return
 
-        base_tags = self._config.base_tags + resource_tags
+        base_tags = []
```

Comment:
We are now doing the same thing in three different places in our code. Have you considered moving this into a single method so as not to repeat code?

Reply:
Each of the three places uses slightly different logic:

1. In [submit_metrics_callback](https://github.com/DataDog/integrations-core/blob/master/vsphere/datadog_checks/vsphere/vsphere.py#L442), we add tags in the excluded_host_tags list **only if the config option is set** (we do nothing otherwise).
2. In [submit_external_host_tags](https://github.com/DataDog/integrations-core/blob/master/vsphere/datadog_checks/vsphere/vsphere.py#L614), we assume that the config option is set, but we only include tags that **are not** in the excluded_host_tags list.
3. In [submit_property_metrics](https://github.com/DataDog/integrations-core/pull/18601/files/31a12560eaaa7ed5af726ce044ec98c18161f701#diff-8d730e4ffb87a6578a9413e77ef494317dcd057e8bcd88d3ae5a50a7629588c6R907),  I add tags in the excluded_host_tags list if the config option is set, or if not, **I add all of the tags**.

Thus, since each place has different conditions, I would have to add at least 2 flags to control how the single method would be used for each place:
```
    def filter_excluded_host_tags(self, potential_tags, excluded_host_tag_configured=True, remove_excluded_host_tags=False):
        if not self._config.excluded_host_tags:
            if excluded_host_tag_configured:
               return potential_tags
            else:
                return []
        if remove_excluded_host_tags:
            return [t for t in potential_tags if t.split(':', 1)[0] not in self._config.excluded_host_tags]
        else:
            return [t for t in potential_tags if t.split(':', 1)[0] in self._config.excluded_host_tags]
```

As such, I feel like using a common method would make the code a lot more confusing. If one looks at the logic in each place separately, it is very easy to tell what tags will be included and what tags aren't.